domain-services/ifix-es-pipeline/src/main/java/org/egov/ifixespipeline/consumers/IfixElasticSearchPipelineListener.java

```diff
@@ -0,0 +1,116 @@
+     * @param topic
+     */
+    @KafkaListener(topics = { "${fiscal.event.kafka.push.topic}", "${fiscal.event.migration.origin.push.topic}"})
+    public void listen(HashMap<String, Object> record, @Header(KafkaHeaders.RECEIVED_TOPIC) String topic) {
+        try {
+            FiscalEventRequest incomingData = objectMapper.convertValue(record, FiscalEventRequest.class);
+
+            // Enrich hierarchy map according to the tenantid encountered by this pipeline to avoid redundant network calls
+            if(!tenantIdVsExpenditureTypeVsUuidsMap.containsKey(incomingData.getFiscalEvent().getTenantId()))
+                tenantIdVsExpenditureTypeVsUuidsMap.put(incomingData.getFiscalEvent().getTenantId(), loadExpenditureTypeVsUuidMap(incomingData.getFiscalEvent().getTenantId()));
+
+            fiscalDataEnrichmentService.enrichFiscalData(incomingData);
+            fiscalDataEnrichmentService.enrichComputedFields(incomingData, tenantIdVsExpenditureTypeVsUuidsMap.get(incomingData.getFiscalEvent().getTenantId()));
+
+            producer.push(indexFiscalEventsTopic, incomingData);
+        }catch(Exception e) {
+            log.error("Exception while reading from the queue: ", e);
+        }
+    }
+
+    private HashMap<String, HashSet<String>> loadExpenditureTypeVsUuidMap(String tenantId){
+        COASearchCriteria criteria = COASearchCriteria.builder().tenantId(tenantId).build();
+        COASearchRequest request = COASearchRequest.builder().requestHeader(new RequestHeader()).criteria(criteria).build();
+        Object result = serviceRequestRepository.fetchResult(getIfixMasterDataUri(), request);
+        COAResponse response = objectMapper.convertValue(result, COAResponse.class);
+        HashMap<String, HashSet<String>> expenditureTypeVsUuidsMap = new HashMap<>();
+        expenditureTypeVsUuidsMap.put("Others", new HashSet<>());
+        expenditureTypeVsUuidsMap.put(electricityCoaHeadName, new HashSet<>());
+        expenditureTypeVsUuidsMap.put(operationsCoaHeadName, new HashSet<>());
+
+        response.getChartOfAccounts().forEach(chartOfAccount -> {
+
+            if(expenditureTypeVsUuidsMap.containsKey(electricityCoaHeadName))
+                expenditureTypeVsUuidsMap.get(electricityCoaHeadName).add(chartOfAccount.getId());
+            else if(expenditureTypeVsUuidsMap.containsKey(operationsCoaHeadName))
+                expenditureTypeVsUuidsMap.get(operationsCoaHeadName).add(chartOfAccount.getId());
+            else if(expenditureTypeVsUuidsMap.containsKey(salaryCoaHeadName))
+                expenditureTypeVsUuidsMap.get(salaryCoaHeadName).add(chartOfAccount.getId());
+            else
+                expenditureTypeVsUuidsMap.get("Others").add(chartOfAccount.getId());
+
+        });
+        return expenditureTypeVsUuidsMap;
+    }
+
+    private StringBuilder getIfixMasterDataUri() {
+        StringBuilder uri = new StringBuilder(ifixMasterDataServiceHost);
+        uri.append(ifixMasterDataServiceSearchEndpoint);
```

Comment:
This isn't a common master data search endpoint. 
This is a COA search endpoint, so it is better to name the method and variables that include `coa`.

Reply:
minor change, to be taken up as part of code cleanup.